src/elements/content-sidebar/SidebarPanels.js

```diff
@@ -68,123 +67,161 @@ const LoadableVersionsSidebar = SidebarUtils.getAsyncSidebarContent(
     MARK_NAME_JS_LOADING_VERSIONS,
 );
 
-const SidebarPanels = ({
-    activitySidebarProps,
-    currentUser,
-    detailsSidebarProps,
-    file,
-    fileId,
-    getPreview,
-    getViewer,
-    hasActivity,
-    hasDetails,
-    hasMetadata,
-    hasSkills,
-    hasVersions,
-    isOpen,
-    metadataSidebarProps,
-    onVersionChange,
-    onVersionHistoryClick,
-    refreshIdentity,
-    versionsSidebarProps,
-}: Props) =>
-    isOpen && (
-        <Switch>
-            {hasSkills && (
-                <Route
-                    exact
-                    path={`/${SIDEBAR_VIEW_SKILLS}`}
-                    render={() => (
-                        <LoadableSkillsSidebar
-                            key={file.id}
-                            file={file}
-                            getPreview={getPreview}
-                            getViewer={getViewer}
-                            startMarkName={MARK_NAME_JS_LOADING_SKILLS}
+class SidebarPanels extends React.Component<Props, State> {
```

Comment:
I converted to a class so I could use `createRef`

Reply:
I don't think `State` is defined in this file. Does this pass a flow check?

Reply:
Ah nope it didnt pass, will fix